### DISCUSSION TITLE
nhibernate async/await pattern

### DISCUSSION THREAD

**Discussion by stcom77** [nhibernate/nhibernate-core#2803] [Q&A]
Hi. I saw the examples how to use nHibernate 5 async features, something like this.
```
async Task<Item> GetItems(){
  return await _session.ListAsync();
}
```
Why should to use async/await instead of just?
```
Task<Item> GetItems(){
  return _session.ListAsync();
}
```
It lets us avoid the async/await overhead. Or maybe are any other reasons to do like in the first example. 
I have read the documentation but didn't find an answer. Maybe are any limitations and the second example will not work?

**Comment by hazzik** [comment]
It does not matter as long as you do not forget to await the final call like this:

```
var items = await GetItems();
```

**Comment by cd21h** [comment]
@stcom77 
I really like explanation given here https://www.reddit.com/r/csharp/comments/9igzbf/whats_the_difference_between_awaiting_a_task_vs/

If you want more technical details check https://blog.stephencleary.com/2016/12/eliding-async-await.html
